lib/Indexer/Tests/Adapter/ReferenceFinder/IndexedReferenceFinderTest.php

```diff
@@ -206,17 +250,35 @@ public function provideMembers(): Generator
                 <?php namespace Bar; $foo = new Foobar(); $foo->b<>ar();
 
                 // File: project/subject1.php
-                <?php namespace Bar; $foo = new Barfoo(); $foo->bar();
+                <?php namespace Bar; $bar = new Barfoo(); $bar->bar();
 
                 // File: project/class1.php
-                <?php namespace Bar; class Foobar extends Barfoo { public function bar() {}}
+                <?php namespace Bar; class Barfoo { public function bar() {}}
 
                 // File: project/class2.php
-                <?php namespace Bar; class Barfoo { public function bar() {}}
+                <?php namespace Bar; class Foobar extends Barfoo { public function bar() {}}
+
+                EOT
+        ,
+            1, 2
+        ];
 
+        yield 'deep members no override' => [
+            <<<'EOT'
+                // File: project/subject.php
+                <?php $foo = new Foobar(); $foo->met<>hod();
+
+                // File: project/subject1.php
+                <?php $bar = new ParentalClass(); $bar->method();
+
+                // File: project/class1.php
+                <?php class ParentalClass { public function method() {}}
+
+                // File: project/class2.php
+                <?php class Foobar extends ParentalClass { }
                 EOT
         ,
-            2, 4 // total number is multipled due to implementation recursion
+            2, 4 // total number is multiple due to implementation recursion
         ];
 
```

Comment:
```php
        yield 'deep members no override' => [
            <<<'EOT'
                // File: project/subject.php
                <?php $foo = new Foobar(); $foo->method();
                // File: project/subject1.php
                <?php $bar = new ParentalClass(); $bar->met<>hod();
                // File: project/class1.php
                <?php class ParentalClass { public function method() {}}
                // File: project/class2.php
                <?php class Foobar extends ParentalClass { }
                EOT
        ,
            ?, ? // 
        ];
```
I have no opinion whether we should include overridden classes or not but probably we should test such case too.

Reply:
There is already a test for that. (The one you posted). Or what do you mean exactly?

Reply:
We have four test cases here. In all of them the cursor position is on method defined in subclass. That's why I thought to check references of some parent method too unless it is out of the scope of this change.

Reply:
Which four test cases are you thinking of? Like having the cursor on the method definition?

Reply:
https://github.com/phpactor/phpactor/pull/2261/files#diff-fe706911329feb79e7b2e1f0a11109e1518c601584a9319a377ab8458f3857abR173
https://github.com/phpactor/phpactor/pull/2261/files#diff-fe706911329feb79e7b2e1f0a11109e1518c601584a9319a377ab8458f3857abR192
https://github.com/phpactor/phpactor/pull/2261/files#diff-fe706911329feb79e7b2e1f0a11109e1518c601584a9319a377ab8458f3857abR250
https://github.com/phpactor/phpactor/pull/2261/files#diff-fe706911329feb79e7b2e1f0a11109e1518c601584a9319a377ab8458f3857abR269

Reply:
You mean that we don't test that if a method is overriden it still is able to find references? That's a good point. I mean it shouldn't be a problem. But I can add a test case for that as well.

Reply:
Yes, but at the moment I am still not sure what to expect from such test.

Reply:
In other IDEs :tm: you can select if you want to see the usage of the base method or the overriden methods. But this is definitely outside the scope of the PR.